app/models/agents/post_agent.rb

```diff
@@ -162,22 +171,29 @@ def normalize_response_headers(headers)
       }
     end
 
-    def handle(data, payload = {}, headers)
-      url = interpolated(payload)[:post_url]
+    def handle(data, event = Event.new, headers)
+      url = interpolated(event.payload)[:post_url]
 
       case method
       when 'get', 'delete'
         params, body = data, nil
       when 'post', 'put', 'patch'
         params = nil
 
-        case (content_type = interpolated(payload)['content_type'])
+        content_type = if has_file_pointer?(event)
```

Comment:
I think this would be slightly more readable as:

```ruby
if has_file_pointer?(event)
  data[interpolated(event.payload)['upload_key'].presence || 'file'] = get_upload_io(event)
  content_type = nil
else
  content_type = interpolated(event.payload)['content_type']
end
```